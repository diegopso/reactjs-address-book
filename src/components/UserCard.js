import React from 'react'
import { Col, Card, Typography, Avatar } from 'antd'
import { upperCaseFirst } from '../helpers/string'

const { Meta } = Card
const { Text } = Typography

const UserCard = ({ user, onClick }) => {
  if (user.hide) {
    return <></>
  }

  user.fullName = upperCaseFirst(user.name.first) + ' ' + upperCaseFirst(user.name.last)
  return (
    <Col className="user-container" span="6">
      <Card onClick={ onClick }>
        <Meta
          avatar={<Avatar src={user.picture.thumbnail} />}
          title={
            <div>
              <Text>{user.fullName}</Text>
              <Text disabled className="text-small">{' @' + user.login.username}</Text>
            </div>
          }
          description={
            <Text>{user.email}</Text>
          }
        />
      </Card>
    </Col>
  )
}

export default UserCard
