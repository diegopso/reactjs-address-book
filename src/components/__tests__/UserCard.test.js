import React from 'react'
import { describe, it, expect } from '@jest/globals'
import { shallow } from 'enzyme'
import UserCard from '../UserCard'

describe('Testing UserCard component:', () => {
  const userMock = {
    email: 'brad.gibson@example.com',
    name: {
      first: 'brad',
      last: 'gibson'
    },
    login: {
      username: 'silverswan131'
    },
    picture: {
      thumbnail: 'https://randomuser.me/api/portraits/thumb/men/75.jpg'
    }
  }

  it('Renders correctly.', () => {
    const wrapper = shallow(<UserCard user={userMock} />)
    expect(wrapper).toMatchSnapshot()
  })
})
