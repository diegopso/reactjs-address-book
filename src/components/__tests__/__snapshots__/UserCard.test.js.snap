// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UserCard component: Renders correctly. 1`] = `
<Col
  className="user-container"
  span="6"
>
  <Card>
    <Meta
      avatar={
        <Avatar
          shape="circle"
          size="default"
          src="https://randomuser.me/api/portraits/thumb/men/75.jpg"
        />
      }
      description={
        <Text>
          brad.gibson@example.com
        </Text>
      }
      title={
        <div>
          <Text>
            Brad Gibson
          </Text>
          <Text
            className="text-small"
            disabled={true}
          >
             @silverswan131
          </Text>
        </div>
      }
    />
  </Card>
</Col>
`;
